.github/workflows/run_test_windows.yml

```diff
@@ -0,0 +1,287 @@
+name: Run Test (Windows WSL)
+
+on:
+    workflow_call:
+        inputs:
+            directories:
+                required: true
+                type: string
+            exclude-dirs:
+                required: false
+                type: string
+                default: ''
+            link-azle:
+                required: true
+                type: boolean
+            run-experimental:
+                required: false
+                type: boolean
+                default: false
+            fuzz:
+                required: false
+                type: boolean
+                default: false
+            call-delay:
+                required: false
+                type: string
+                default: '.1'
+            time-limit:
+                required: false
+                type: string
+                default: '300'
+
+jobs:
+    get-test-infos:
+        name: Get test infos
+        runs-on: ubuntu-latest
+        outputs:
+            test-infos: ${{ steps.get-test-infos.outputs.test-infos }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: get-test-infos
+              uses: ./.github/actions/get_test_infos
+              with:
+                  directories: ${{ inputs.directories }}
+                  exclude-dirs: ${{ inputs.exclude-dirs }}
+
+    run-test:
+        name: '${{matrix.test.name}} | ${{matrix.test.displayPath}} | ${{matrix.azle_source}} | WSL'
+        needs: get-test-infos
+        runs-on: windows-latest
+        env:
+            ETHEREUM_URL: ${{ secrets.ETHEREUM_URL }}
+            AZLE_IDENTITY_STORAGE_MODE: 'plaintext'
+            AZLE_END_TO_END_TEST_LINK_AZLE: ${{ matrix.azle_source == 'repo' }}
+
+        strategy:
+            fail-fast: false # We want to see which example tests succeed and which ones fail, we don't want one example test to cancel the rest
+            matrix: # spins up one job per combination of test and code source (repo or npm).
+                azle_source:
+                    - ${{ inputs.link-azle == true && 'repo' || 'npm' }}
+                test: ${{ fromJSON(needs.get-test-infos.outputs.test-infos) }}
+        steps:
+            - name: Setup WSL
+              uses: Vampire/setup-wsl@v3
+              with:
+                  distribution: Ubuntu-22.04
+                  update: 'true'
+                  additional-packages: curl build-essential git jq
+                  wsl-conf: |
+                      [boot]
+                      systemd=true
+                  set-as-default: 'true'
+                  wsl-shell-user: 'root'
```

Comment:
I'm not sure this is the right choice. Our users will not be running as root. We should create a regular user with regular user permissions.